contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -99,121 +99,168 @@ contract DisputeManagerFacet is StateChannelCommon {
                 .executeStateTransitionOnState(channelId, encodedState, _tx);
     }
 
-    function getDispute(
-        bytes32 channelId
-    ) public view returns (Dispute memory) {
-        return disputes[channelId];
+    
+
+    function _handleBlockFraudProofs(
+       
+    ) internal {
+      
+       
+
+    }
+
+    function _handleDisputeFraudProofs(
+        // core specific checks
+        // apply proofs
+        // apply dispute to latest state
+        // set state
+    ) internal {
+
+    }
+
+    function _handleTimeoutDispute(
+
+    ) internal {
+
     }
 
     function createDispute(
         bytes32 channelId,
         uint forkCnt,
-        bytes memory encodedLatestFinalizedState,
-        bytes memory encodedLatestCorrectState,
-        ConfirmedBlock[] memory virtualVotingBlocks,
-        address timedoutParticipant,
-        uint foldedTransactionCnt,
-        Proof[] memory proofs
-    ) public {
-        require(!isDisputeInProgress(channelId), ErrorDisputeInProgrees());
-        require(getForkCnt(channelId) == forkCnt, ErrorDisputeForkMismatch());
+        bytes calldata encodedLatestState,
+        StateProof[] calldata stateProofs,
+        address[] calldata onChainSlashes,
+        Proof[] calldata proofs
+    ) public { 
+        // sanity checks
+        if(onChainSlashes.length != onChainSlashedParticipants.length 
+        || onChainSlashes[onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
+            revert("onChainSlashes does not match onChainSlashedParticipants");
+        }
+        if(proofs.length == 0) return;
+
+        // verify latest state
         address[] memory participants = getParticipants(channelId, forkCnt);
-        bool isParticipant = StateChannelUtilLibrary.isAddressInArray(
-            participants,
-            msg.sender
-        );
-        require(isParticipant, ErrorNotParticipant());
+        // check if its genesis
+        if(stateProofs.length == 0) {
+           bool isGenesis = isGenesisState(channelId, forkCnt, encodedLatestState); 
+           if(!isGenesis) {
+            revert("latest state is not valid");
+           }
+        }else{
+            _verifyStateProof(encodedLatestState, stateProofs, participants);
```

Comment:
All of this logic is not done here - it's done in auditing which is 'stateless', but has synchronous composability to read data from state (not modify it) - explained in the doc

Here we just commit to the dispute struct and optionally invoke auditing which returns true/false and other things defined in the docs

Reply:
but we are. checking if the disputer has submitted sensible latest state?

Reply:
Unless we invoke auditing on-chain (on-chain execution) we don't check anything apart the safety guards. In general we just try to commit and allow others to audit off-chain since it's free and they have the full state and can check. If we want to do on-chain execution we have to upload more than the minimum requirement (e.g. A hash is enough to commit to a state, but you can't transition the state from it. Since peers off-chain have the full state it's enough for them to check, but if you want to invoke on-chain execution you have to upload the state which can be easily verified against the commitment). 

So all of this is part of auditing which requires more data - I'll explain tomorrow in the call 

We should have calls more frequently